lib/ui/window/platform_configuration.cc

```diff
@@ -377,7 +377,18 @@ void PlatformConfiguration::BeginFrame(fml::TimePoint frameTime,
   }
   tonic::DartState::Scope scope(dart_state);
 
-  int64_t microseconds = (frameTime - fml::TimePoint()).ToMicroseconds();
+  int64_t microseconds = frameTime.ToEpochDelta().ToMicroseconds();
+
+  static int64_t last_microseconds = 0;
+  if (last_microseconds > microseconds) {
+    // Do not allow time traveling frametimes
+    // github.com/flutter/flutter/issues/106277
+    FML_LOG(ERROR)
+        << "Reported frame time is older than the last one; clamping. "
+        << microseconds << " < " << last_microseconds
+        << " ~= " << last_microseconds - microseconds;
+    microseconds = last_microseconds;
+  }
```

Comment:
Simplify? The static `last_microseconds` looks like it's supposed to be tracking some state over time, but it's not.
```suggestion
  if (microseconds < 0) {
    // Do not allow time traveling frametimes
    // github.com/flutter/flutter/issues/106277
    FML_LOG(ERROR)
        << "Frame delta was negative (" << microseconds
        << "). Clamping to zero.";
    microseconds = 0;
  }
```

Reply:
I failed to set `last_microseconds = microseconds`; which is odd because it was in my patch somewhere else in the code. I chose to pull it up higher since this affects iOS and Android apparently (recent bug report).